packages/core/src/baseCogniteClient.ts

```diff
@@ -235,46 +263,66 @@ export default class BaseCogniteClient {
    *
    * // using Cognite authentication flow
    * client.loginWithOAuth({
-   *   project: '[PROJECT]',
-   *   onAuthenticate: REDIRECT // optional, REDIRECT is by default
+   * type: 'CDF_OAUTH',
+   * options: {
+   * project: '[PROJECT]',
+   * onAuthenticate: REDIRECT // optional, REDIRECT is by default
+   * }
    * });
    *
    * // or you can sign in using AzureAD authentication flow (in case your projects supports it)
    * client.loginWithOAuth({
-   *   cluster: '[CLUSTER]',
-   *   clientId: '[CLIENT_ID]', // client id of your AzureAD application
-   *   tenantId: '[TENANT_ID]', // tenant id of your AzureAD tenant. Will be set to 'common' if not provided
+   * type: 'AAD_OAUTH',
+   * options: {
+   * cluster: '[CLUSTER]',
+   * clientId: '[CLIENT_ID]', // client id of your AzureAD application
+   * tenantId: '[TENANT_ID]', // tenant id of your AzureAD tenant. Will be set to 'common' if not provided
+   * }
    * });
    *
    * // you also have ability to sign in using ADFS
    * client.loginWithOAuth({
-   *   authority: https://example.com/adfs/oauth2/authorize,
-   *   requestParams: {
-   *     cluster: 'cluster-name',
-   *     clientId: 'adfs-client-id',
-   *   },
+   * type: 'ADFS_OAUTH',
+   * options: {
+   * authority: https://example.com/adfs/oauth2/authorize,
+   * requestParams: {
+   * cluster: 'cluster-name',
+   * clientId: 'adfs-client-id',
+   * },
+   * }
+   * });
+   *
+   * // or with client credentials
```

Comment:
this is not a part of this PR but will be a part of my PR. possibly remove.

Reply:
done